### PR TITLE
[JENKINS-42213] Changes in POM

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-42213](https://issues.jenkins-ci.org/browse/JENKINS-42213)

- Update baseline according to plugin dependencies.
- Check java.level is set to correct version.
- Bump and include plugins to fix PCT for 2.32.2

@reviewbybees @stephenc

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -21,8 +21,9 @@
     </licenses>
 
     <properties>
-        <jenkins.version>1.609.3</jenkins.version>
-        <workflow.version>1.14</workflow.version>
+        <jenkins.version>1.625.3</jenkins.version>
```
Comment (varyvol) [inline pom.xml]: display-url-api has this version as baseline.

File: pom.xml [hunk 2]
```diff
@@ -21,8 +21,9 @@
...
+        <java.level>7</java.level>
```
Comment (jglick) [inline pom.xml]: You may just delete this as it is the default.

Comment (varyvol) [inline pom.xml]: @jglick I personally prefer putting it as it makes easier to distinguish it with a quick sight. But if you prefer not having it, I am not against removing it either.

Comment (stephenc) [inline pom.xml]: If it is not needed, it is dead code. Delete delete delete

Comment (varyvol) [inline pom.xml]: Removed then.

**Pull request merged**